src/plugins/condition/ConditionManager.js

```diff
@@ -0,0 +1,254 @@
+            return this.openmct.objects.makeKeyString(conditionConfigurationId);
+        });
+        let newConditionIdentifiers = newDomainObject.configuration.conditionCollection.map((conditionConfigurationId) => {
+            return this.openmct.objects.makeKeyString(conditionConfigurationId);
+        });
+
+        this.domainObject = newDomainObject;
+
+        //check for removed conditions
+        oldConditionIdentifiers.forEach((identifier, index) => {
+            if (newConditionIdentifiers.indexOf(identifier) < 0) {
+                this.removeCondition(identifier);
+            }
+        });
+
+        let newConditionCount = this.domainObject.configuration.conditionCollection.length - this.conditionCollection.length;
+
+        for (let i = 0; i < newConditionCount; i++) {
+            let conditionConfigurationId = this.domainObject.configuration.conditionCollection[i];
+            this.openmct.objects.get(conditionConfigurationId).then((conditionConfiguration) => {
+                this.initCondition(conditionConfiguration, i);
+            });
+        }
+
+    }
+
+    initCondition(conditionConfiguration, index) {
+        let condition = new Condition(conditionConfiguration, this.openmct);
+        condition.on('conditionResultUpdated', this.handleConditionResult.bind(this));
+        if (index !== undefined) {
+            this.conditionCollection.splice(index + 1, 0, condition);
+        } else {
+            this.conditionCollection.unshift(condition);
+        }
+        //There are no criteria for a default condition and hence no subscriptions.
+        //Hence the conditionResult must be manually triggered for it.
+        if (conditionConfiguration.isDefault) {
+            this.handleConditionResult();
+        }
+    }
+
+    createConditionDomainObject(isDefault, conditionConfiguration) {
+        let conditionObj;
+        if (conditionConfiguration) {
+            conditionObj = {
+                ...conditionConfiguration,
+                name: `Copy of ${conditionConfiguration.name}`,
+                identifier: {
+                    ...this.domainObject.identifier,
+                    key: uuid()
+                }
+            };
+        } else {
+            conditionObj = {
+                isDefault: isDefault,
+                type: 'condition',
+                name: isDefault ? 'Default' : 'Unnamed Condition',
+                identifier: {
+                    ...this.domainObject.identifier,
+                    key: uuid()
+                },
+                configuration: {
+                    name: isDefault ? 'Default' : 'Unnamed Condition',
+                    output: 'false',
+                    trigger: 'any',
+                    criteria: isDefault ? [] : [{
+                        telemetry: '',
+                        operation: '',
+                        input: '',
```

Comment:
Shouldn't criteria.input be initialized as an empty array? If inputCount for the operation is 0 or one it still works fine as an array and this way we don't have to check for typeof Array in telemetryCriterion class.

Reply:
Yes, that is correct. The state of things as they were at the time I made this code change did not have input as an array. I will change it.

Reply:
done.